feature-libs/order/core/facade/order-history-extended-view.service.ts

```diff
@@ -0,0 +1,260 @@
+                  return orderView;
+                })
+              );
+            } else {
+              orderView.consignments?.push(consignmentView);
+              return of(orderView);
+            }
+          }
+        );
+        if (requests === undefined || requests.length < 1) {
+          return of(orderView);
+        }
+        return combineLatest(requests).pipe(
+          switchMap((orders: OrderView[] | undefined) => {
+            if (orders !== undefined) {
+              return of(orders[0]);
+            } else {
+              return of(order);
+            }
+          })
+        );
+        //<-----------------
+      })
+    );
+  }
+
+  getOrderListWithDetails(
+    pageSize: number
+  ): Observable<OrderHistoryListView | undefined> {
+    let orderListView: OrderHistoryListView = {};
+    return this.orderHitsoryService.getOrderHistoryList(pageSize).pipe(
+      switchMap((orderList: OrderHistoryList | undefined) => {
+        orderListView.pagination = orderList?.pagination;
+        orderListView.sorts = orderList?.sorts;
+        orderListView.orders = [];
+        const requests = (orderList?.orders ?? []).map(
+          (order: OrderHistory) => {
+            let orderView: OrderHistoryView = { ...order };
+            return this.getOrderDetailsWithTracking(order?.code ?? '').pipe(
+              map((orderDetail: OrderView | undefined) => {
+                /** filling extra fields ---> */
+                orderView.entries = orderDetail?.entries;
+                orderView.consignments = orderDetail?.consignments;
+                orderView.unconsignedEntries = orderDetail?.unconsignedEntries;
+                orderView.returnRequests = [];
+                /** filling extra fields <--- */
+                orderListView.orders?.push(orderView);
+                return orderListView;
+              })
+            );
+          }
+        );
+        return combineLatest(requests);
+      }),
+      map((requests: OrderHistoryListView[] | undefined) => {
+        if (requests !== undefined) {
+          return requests[0];
+        } else {
+          return {};
+        }
+      })
+    );
+  }
+
+  getOrderHistoryList(
+    pageSize: number
+  ): Observable<OrderHistoryListView | undefined> {
+    const orderHistoryListRequest = this.getOrderListWithDetails(pageSize);
+    const returnRequestListRequest =
+      this.orderReturnRequestService.getOrderReturnRequestList();
+    return combineLatest([
+      orderHistoryListRequest,
+      returnRequestListRequest,
+    ]).pipe(
+      switchMap(
+        (
+          responses: [
+            OrderHistoryListView | undefined,
+            ReturnRequestList | undefined
+          ]
+        ) => {
+          const returnRequests = responses?.[1]?.returnRequests;
+          const orderHistory = responses?.[0];
+          if (returnRequests && orderHistory?.orders) {
+            console.log(returnRequests);
```

Comment:
[MINOR]
Let's remove debugging console.logs from the PR.

Please review the whole PR and remove other `console` usages, if present.